src/hooks/useWatchBundle.tsx

```diff
@@ -0,0 +1,61 @@
+import {
+  iterateEach,
+  makeCastingSpec,
+  makeFollower,
+  makeLeader,
+} from "@agoric/casting";
+import { toast } from "react-toastify";
+import { useNetwork } from "./useNetwork";
+import { BundleFollowerToastMessage } from "../components/BundleFollowerToastMessage";
+
+type IteratorEnvelope = {
+  value?: {
+    endoZipBase64Sha512: string;
+    installed: boolean;
+    error: unknown;
+  };
+  error?: { message: string; stack: string };
+};
+
+export const useWatchBundle = () => {
+  const { networkConfig } = useNetwork();
+  const leader = networkConfig ? makeLeader(networkConfig.rpc) : undefined;
+
+  const watchBundle = async (
+    expectedEndoZipBase64Sha512: string,
+    { height }: { height: number },
+  ) => {
+    if (!leader) throw Error("Unexpected error: leader not found.");
+    const castingSpec = makeCastingSpec(":bundles");
+    const follower = makeFollower(castingSpec, leader);
+    for await (const envelope of iterateEach(follower, { height })) {
+      const { value, error } = envelope as IteratorEnvelope;
+      if (!value && error) {
+        toast.error(`Bundle installation failed.\nSee console for details.`);
+        console.log(envelope);
+        throw error;
+      }
+      if (value) {
+        const { endoZipBase64Sha512, installed, error } = value;
```

Comment:
I had trouble determining the best way to simulate error cases - I took out a random segment in `endoZipBase64` from a working bundle.

I observed `error` coming from `envelope`, unlike [this example](https://github.com/Agoric/agoric-sdk/blob/master/packages/agoric-cli/src/publish.js#L316-L328) where its expected it to come from `envelope.value`.  `value` (and `endoZipBase64Sha512 `) are undefined. This is potentially a result of my testing approach.

Reply:
cc @dckc